wolfSSL/src/appendix07.md

```diff
@@ -954,3 +954,128 @@ Technical documentation and other resources such as known answer tests can be fo
+- height: height of an individual Merkle tree.
+- Winternitz: number of bits from hash used in a Winternitz chain. Used as a space-time tradeoff for the signature size.
+
+wolfSSL supports all LMS/HSS parameters defined in RFC8554:
+
+- levels = {1..8}
+- height = {5, 10, 15, 20, 25}
+- Winternitz = {1, 2, 4, 8}
+
+The number of available signatures is:
+- N = 2 ** (levels * height)
+
+For convenience some parameter sets have been predefined in the enum `wc_LmsParm`. Its values are shown in the table below:
+
+parameter set |  description
+--------------------  |  ---------------------------
+WC_LMS_PARM_NONE      | Not set, use default (WC_LMS_PARM_L1_H15_W2)
+WC_LMS_PARM_L1_H15_W2 | 1 level Merkle tree of 15 height, Winternitz 2
+WC_LMS_PARM_L1_H15_W4 | same as above, Winternitz 4
+WC_LMS_PARM_L2_H10_W2 | 2 level Merkle tree of 10 height, Winternitz 4
+WC_LMS_PARM_L2_H10_W4 | same as above, Winternitz 4
+WC_LMS_PARM_L2_H10_W8 | same as above, Winternitz 8
+WC_LMS_PARM_L3_H5_W2  | 3 level Merkle tree of 5 height, Winternitz 2
+WC_LMS_PARM_L3_H5_W4  | same as above, Winternitz 4
+WC_LMS_PARM_L3_H5_W8  | same as above, Winternitz 8
+WC_LMS_PARM_L3_H10_W4 | 3 level Merkle tree of 10 height, Winternitz 4
+WC_LMS_PARM_L4_H5_W8  | 4 level Merkle tree of 5 height, Winternitz 8
+
+The signature size and number of signatures is shown with respect to the parameter set here:
+
+parameter set |  signature size | number of signatures
+--------------------  | -------------------- | --------------------
+WC_LMS_PARM_L1_H15_W2 | 4784 | 32768
+WC_LMS_PARM_L1_H15_W4 | 2672 | 32768
+WC_LMS_PARM_L2_H10_W2 | 9300 | 1048576
+WC_LMS_PARM_L2_H10_W4 | 5076 | 1048576
+WC_LMS_PARM_L2_H10_W8 | 2964 | 1048576
+WC_LMS_PARM_L3_H5_W2  | 13496 | 32768
+WC_LMS_PARM_L3_H5_W4  | 7160 | 32768
+WC_LMS_PARM_L3_H5_W8  | 3992 | 32768
+WC_LMS_PARM_L3_H10_W4 | 7640 | 1073741824
+WC_LMS_PARM_L4_H5_W8  | 5340 | 1048576
+
+As can be seen from the tables, signature sizes are primarily determined by the levels and Winternitz parameters, and height to a lesser extent:
+- Larger levels values increase signature size significantly.
+- Larger height values increase signature size modestly.
+- Larger winternitz values will reduce the signature size, at the expense of longer key generation and sign/verify times.
+
+Key generation time is strongly determined by the height of the first level tree. A 3 level, 5 height tree is much faster than 1 level, 15 height at initial key gen, even if the number of available signatures is the same.
+
+### LMS/HSS Build Instructions
+
+Please see the wolfSSL repo's INSTALL file (https://github.com/wolfSSL/wolfssl/blob/master/INSTALL). Item 17 (Building with hash-sigs lib for LMS/HSS support [EXPERIMENTAL]) has instructions on how to configure and build wolfSSL and the hash-sigs LMS/HSS library.
+
+#### Benchmark Data
+
+The following benchmark data was taken on an 8-core Intel i7-8700 CPU @ 3.20GHz, on Fedora 38 (`6.2.9-300.fc38.x86_64`).
```

Comment:
State how many threads were used.

Reply:
You didn't mention how many threads were spawned in this line.

Reply:
I described it in the multi-threaded and single-threaded test results below.

When using `hss_lib_thread.a` the hash-sigs lib will spawn up to 16 worker threads, depending on the parameters  used and what it's calculating. This is outside our control, unless we use their API to set the maximum for number of worker threads. In practice I only saw 6 cores utilized during key gen.